Copernicus grid codes for build_renewable_profiles.py

euronion opened the pull request:
Partially addresses: https://github.com/pypsa-meets-africa/pypsa-africa/issues/108

## Changes proposed in this Pull Request

With this PR, the land use grid codes according to the Copernicus Land Cover are implemented.
Land types used for RES area potential evaluation are comparable to those used in PyPSA-EUR with the CORINE (Europe only) dataset.


## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Newly introduced dependencies are added to `envs/environment.yaml` and `envs/environment.docs.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`, `config.tutorial.yaml`, and `test/config.test1.yaml`.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [x] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes.

euronion commented:
CI is failing during plotting (linux/mac) / retrieval of databundle (windows). Doesn't seem related to this PR. Are there issues with the CI at the moment?

davide-f commented:
> CI is failing during plotting (linux/mac) / retrieval of databundle (windows). Doesn't seem related to this PR. Are there issues with the CI at the moment?

Errors in the plotting sometimes occurr but usually not related to our workflow, sometimes the plotting package is not able to download some necessary files and turns an error.
Instead, there shouldn't be errors on retrieve_databundle.
On my side, the plotting is not a critical part in verifying if the workflow works fine, but the previous part should work.
I rerun the workflow to better verify the issue

euronion commented:
Board is all green, ready to merge (on my side).

euronion commented:
Nearly forgot the release notes. 😊

davide-f commented:
Thanks!
Did you test also the non-tutorial version locally?

euronion commented:
Yes I did (only "NG", January only, 3H resolution) without issues.

davide-f commented:
Awesome, last point is, may you improve the documentation/comment related to the choice of the values?

It is not clear to me the values you chose for grid codes, values are pretty high while the classes of the dataset are up to 23; see technical folder at https://land.copernicus.eu/global/products/lc

pz-max commented:
@euronion is always there and surprising people. Great contribution! Thanks Davide for guiding Johannes :100:

euronion commented:
> Awesome, last point is, may you improve the documentation/comment related to the choice of the values?
> 
> It is not clear to me the values you chose for grid codes, values are pretty high while the classes of the dataset are up to 23; see technical folder at https://land.copernicus.eu/global/products/lc

Sure! I've updated the documentation and added a few comments to the config at one point (not everywhere).

To directly address your question:
The grid codes are hierarchical, not ordinal with layers 1 to 3. Each digit represents a layer.

E.g.

11X represent Closed forests, with 111-116 specifying subtypes of closed forests
12X represent open forests, with 121-126 specifying subtypes of closed forests.

That's why the numbers for the map codes seem so high, despite there only 23 types being distinguished.

the list of codes is documented in the PUM here: https://land.copernicus.eu/global/sites/cgls.vito.be/files/products/CGLOPS1_PUM_LC100m-V3_I3.4.pdf

![image](https://user-images.githubusercontent.com/42553970/154858517-897cb797-920b-482f-a553-d7a1452c498f.png)

davide-f commented:
@euronion awesome! I'll wait CI to end and I'll merge everything whatever the result.
Good job and explanation!

pz-max commented on scripts/build_renewable_profiles.py:
```diff
@@ -287,20 +288,22 @@
+                    paths.copernicus,
+                    codes=copernicus["distance_grid_codes"],
+                    buffer=copernicus["distance"],
+                    crs="EPSG:4236",
```
Hey @euronion, I think we have a bug here. CRS should be EPSG:4326 ? Like typically provided: https://zenodo.org/record/3518038#.YoytIznMKEA

euronion commented:
Yes, indeed, 2 and 3 are switched.
